pvlib/irradiance.py

```diff
@@ -2048,3 +2048,70 @@ def _get_dirint_coeffs():
         [0.743440, 0.592190, 0.603060, 0.316930, 0.794390]]
 
     return coeffs[1:, 1:, :, :]
+
+
+def dni(ghi, dhi, zenith, clearsky_dni=None, clearsky_tolerance=1,
```

Comment:
I find the name dni_from_components a bit confusing since components could mean different things. Maybe we just name it after what it does `dni_from_ghi_and_dhi`?